controllers/rollback_handlers.go

```diff
@@ -18,18 +18,90 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"path/filepath"
 
-	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
 	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"github.com/openshift-kni/lifecycle-agent/internal/common"
+	"github.com/openshift-kni/lifecycle-agent/internal/ostreeclient"
+	lcautils "github.com/openshift-kni/lifecycle-agent/utils"
 	ctrl "sigs.k8s.io/controller-runtime"
+
+	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 //nolint:unparam
-func (r *ImageBasedUpgradeReconciler) handleRollback(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+func (r *ImageBasedUpgradeReconciler) startRollback(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+	utils.SetRollbackStatusInProgress(ibu, "Initiating rollback")
+
+	stateroot, err := r.RPMOstreeClient.GetUnbootedStaterootName()
+	if err != nil {
+		utils.SetRollbackStatusFailed(ibu, err.Error())
+		return requeueWithError(err)
```

Comment:
Looks like we want to fail the rollback in this case as you have set the rollback condition to fail, then we should return "doNotRqueue(), nil", otherwise it will keep retrying.
Same comment for other places have called "utils.SetRollbackStatusFailed(ibu, err.Error())"